src/Compilers/VisualBasic/Portable/Symbols/TypeSymbolExtensions.vb

```diff
@@ -288,11 +296,25 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                     Dim container2 As NamedTypeSymbol = DirectCast(t2, NamedTypeSymbol)
 
                     Do
+
+                        If (container1.IsDefinition <> container2.IsDefinition) AndAlso
+                            Not ((compareKind And TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds) <> 0 AndAlso
+                            (DirectCast(container1, NamedTypeSymbol).HasTypeArgumentsCustomModifiers OrElse DirectCast(container2, NamedTypeSymbol).HasTypeArgumentsCustomModifiers)) Then
+
+                            Return False
+                        End If
```

Comment:
It feels like this code should be placed right before the ```Loop``` statement below. The check for the first iteration is performed before we enter the loop.